### PR TITLE
OCM-1109 | fix: Move logic only related to auto into ModeAuto check

### PR DESCRIPTION
**Pull request by hunterkepley**
[OCM-1109](https://issues.redhat.com//browse/OCM-1109) - move all logic that is only needed for auto mode into the `ModeAuto` check to allow the least amount of permissions possible, hopefully no permissions will work with this change, but we need to check for the account role name. So maybe that one permission is needed by default.

### CODE REVIEW

File: cmd/create/operatorroles/by_clusterkey.go [hunk 1]
```diff
@@ -27,68 +27,70 @@ func handleOperatorRoleCreationByClusterKey(r *rosa.Runtime, env string,
...
 
 	switch mode {
 	case aws.ModeAuto:
+		// Check to see if IAM operator roles have already created
+		missingRoles, err := validateOperatorRoles(r, cluster)
+		if err != nil {
+			if strings.Contains(err.Error(), "AccessDenied") {
+				r.Reporter.Debugf("Failed to verify if operator roles exist: %s", err)
+			} else {
+				r.Reporter.Errorf("Failed to verify if operator roles exist: %s", err)
```
Comment (den-rgb) [inline cmd/create/operatorroles/by_clusterkey.go:58]: Single quotes on %s missing

File: cmd/create/operatorroles/by_clusterkey.go [hunk 2]
```diff
@@ -27,68 +27,70 @@ func handleOperatorRoleCreationByClusterKey(r *rosa.Runtime, env string,
...
-
-	hostedCPPolicies := aws.IsHostedCPManagedPolicies(cluster)
 
 	switch mode {
 	case aws.ModeAuto:
+		// Check to see if IAM operator roles have already created
+		missingRoles, err := validateOperatorRoles(r, cluster)
+		if err != nil {
+			if strings.Contains(err.Error(), "AccessDenied") {
+				r.Reporter.Debugf("Failed to verify if operator roles exist: %s", err)
```
Comment (den-rgb) [inline cmd/create/operatorroles/by_clusterkey.go:56]: Single quotes on %s missing

File: cmd/create/operatorroles/by_clusterkey.go [hunk 3]
```diff
@@ -27,68 +27,70 @@ func handleOperatorRoleCreationByClusterKey(r *rosa.Runtime, env string,
...
+			}
+
+			if !args.forcePolicyCreation {
+				r.Reporter.Infof("Operator Roles already exists")
+				return nil
+			}
+		}
+		roleName, err := aws.GetInstallerAccountRoleName(cluster)
+		if err != nil {
+			r.Reporter.Errorf("Expected parsing role account role '%s': %v", cluster.AWS().STS().RoleARN(), err)
```
Comment (den-rgb) [inline cmd/create/operatorroles/by_clusterkey.go:78]: Single quotes on %v missing

File: cmd/create/operatorroles/by_clusterkey.go [hunk 4]
```diff
@@ -27,68 +27,70 @@ func handleOperatorRoleCreationByClusterKey(r *rosa.Runtime, env string,
...
+			os.Exit(1)
+		}
+
+		path, err := aws.GetPathFromAccountRole(cluster, aws.AccountRoles[aws.InstallerAccountRole].Name)
+		if err != nil {
+			r.Reporter.Errorf("Expected a valid path for '%s': %v", cluster.AWS().STS().RoleARN(), err)
```
Comment (den-rgb) [inline cmd/create/operatorroles/by_clusterkey.go:84]: Single quotes on %v missing

File: cmd/create/operatorroles/by_clusterkey.go (hunk 1 above)
Comment (hunterkepley) [reply cmd/create/operatorroles/by_clusterkey.go:58]: i can add that, i didnt make any of this code though so dont blame me :laughing:

Comment (den-rgb) [reply cmd/create/operatorroles/by_clusterkey.go]: Yeah I understand haha , just to keep code similarity across the repo :)

File: cmd/create/operatorroles/by_clusterkey.go [hunk 5]
```diff
@@ -27,80 +27,82 @@ func handleOperatorRoleCreationByClusterKey(r *rosa.Runtime, env string,
...
+		}
+		var accountRoleVersion string
+
 		if !output.HasFlag() || r.Reporter.IsTerminal() {
 			r.Reporter.Infof("Creating roles using '%s'", r.Creator.ARN)
 		}
 		accountRoleVersion, err = r.AWSClient.GetAccountRoleVersion(roleName)
 		if err != nil {
-			r.Reporter.Errorf("Error getting account role version %s", err)
+			r.Reporter.Errorf("Error getting account role version '%s'", err)
```
Comment (ciaranRoche) [inline cmd/create/operatorroles/by_clusterkey.go:99]: can you update to `%v` for the `err` :pray:

File: cmd/create/operatorroles/by_clusterkey.go [hunk 6]
```diff
@@ -137,7 +139,7 @@ func handleOperatorRoleCreationByClusterKey(r *rosa.Runtime, env string,
 		fmt.Println(commands)
 
 	default:
-		r.Reporter.Errorf("Invalid mode. Allowed values are %s", aws.Modes)
+		r.Reporter.Errorf("Invalid mode. Allowed values are '%s'", aws.Modes)
```
Comment (ciaranRoche) [inline cmd/create/operatorroles/by_clusterkey.go:142]: Same as above can you update all these errors to %v please

File: cmd/create/operatorroles/by_clusterkey.go [hunk 7]
```diff
@@ -27,80 +27,82 @@ func handleOperatorRoleCreationByClusterKey(r *rosa.Runtime, env string,
...
-	if err != nil {
-		r.Reporter.Errorf("%s", err)
-	}
-
-	hostedCPPolicies := aws.IsHostedCPManagedPolicies(cluster)
 
 	switch mode {
 	case aws.ModeAuto:
+		// Check to see if IAM operator roles have already created
+		missingRoles, err := validateOperatorRoles(r, cluster)
```
Comment (ciaranRoche) [inline cmd/create/operatorroles/by_clusterkey.go:53]: Why is this not valid for both auto and manual?

Comment (hunterkepley) [reply cmd/create/operatorroles/by_clusterkey.go:53]: @ciaranRoche This uses AWS, as well as is not used in manual

This ticket's goal is to allow creation of the commands with 0 permissions, and this needs GetRole permission

Comment (ciaranRoche) [reply cmd/create/operatorroles/by_clusterkey.go:53]: ack, the 0 permissions makes sense. Thanks Hunter

**Review by ciaranRoche — APPROVED** [review]
/lgtm

lets merge and get it through QE for testing for RC2

**Pull request merged**